Firestore/Source/API/FIRFilter+Internal.h

```diff
@@ -0,0 +1,115 @@
+
+@class FIRFieldPath;
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(Filter)
+@interface FIRFilter : NSObject
+
+#pragma mark - Public Methods
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+                      isEqualTo:(nonnull id)value NS_SWIFT_NAME(whereField(_:isEqualTo:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                          isEqualTo:(nonnull id)value NS_SWIFT_NAME(whereField(_:isEqualTo:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+                   isNotEqualTo:(nonnull id)value NS_SWIFT_NAME(whereField(_:isNotEqualTo:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                       isNotEqualTo:(nonnull id)value NS_SWIFT_NAME(whereField(_:isNotEqualTo:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+                  isGreaterThan:(nonnull id)value NS_SWIFT_NAME(whereField(_:isGreaterThan:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                      isGreaterThan:(nonnull id)value
+    NS_SWIFT_NAME(whereField(_:isGreaterThan:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+         isGreaterThanOrEqualTo:(nonnull id)value
+    NS_SWIFT_NAME(whereField(_:isGreaterOrEqualTo:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+             isGreaterThanOrEqualTo:(nonnull id)value
+    NS_SWIFT_NAME(whereField(_:isGreaterOrEqualTo:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+                     isLessThan:(nonnull id)value NS_SWIFT_NAME(whereField(_:isLessThan:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                         isLessThan:(nonnull id)value NS_SWIFT_NAME(whereField(_:isLessThan:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+            isLessThanOrEqualTo:(nonnull id)value
+    NS_SWIFT_NAME(whereField(_:isLessThanOrEqualTo:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                isLessThanOrEqualTo:(nonnull id)value
+    NS_SWIFT_NAME(whereField(_:isLessThanOrEqualTo:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+                  arrayContains:(nonnull id)value NS_SWIFT_NAME(whereField(_:arrayContains:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                      arrayContains:(nonnull id)value
+    NS_SWIFT_NAME(whereField(_:arrayContains:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+               arrayContainsAny:(nonnull NSArray<id> *)values
+    NS_SWIFT_NAME(whereField(_:arrayContainsAny:));
+
++ (FIRFilter *)filterWhereFieldPath:(nonnull FIRFieldPath *)field
+                   arrayContainsAny:(nonnull NSArray<id> *)values
+    NS_SWIFT_NAME(whereField(_:arrayContainsAny:));
+
++ (FIRFilter *)filterWhereField:(nonnull NSString *)field
+                        inArray:(nonnull NSArray<id> *)values NS_SWIFT_NAME(whereField(_:in:));
```

Comment:
@cherylEnkidu I have updated the API proposal to use `in` and `notIn` (rather than `inArray` and `notInArray` which were proposed before. Please make the update throughout this PR.